tools/clang/lib/AST/ExprConstant.cpp

```diff
@@ -9451,6 +9451,22 @@ bool Expr::isIntegerConstantExpr(llvm::APSInt &Value, const ASTContext &Ctx,
   return true;
 }
 
+bool Expr::isVulkanSpecConstantExpr(const ASTContext &Ctx,
+                                    APValue *Result) const {
+  if (auto *D = dyn_cast<DeclRefExpr>(this)) {
+    if (auto *V = dyn_cast<VarDecl>(D->getDecl())) {
+      if (V->hasAttr<VKConstantIdAttr>()) {
+        if (const Expr *I = V->getAnyInitializer()) {
+          if (!I->isCXX11ConstantExpr(Ctx, Result))
+            return false;
+        }
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
```

Comment:
I might be clearer to require an individual if-statements to reduce the nesting.

```
auto *D = dyn_cast<DeclRefExpr>(this)
if (!D) return false;
auto *V = dyn_cast<VarDecl>(D->getDecl())
if (!V || !V->hasAttr<VkConstantIdAttr>()) return false;
...
```

Reply:
Done.